src/componets/ArticlePage.jsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from "react";
+import { useParams, Link } from "react-router-dom";
+import Comments from "./Comments";
+
+export default function ArticlePage() {
+  const { article_id } = useParams();
+  const [isLoading, setIsLoading] = useState(true);
+  const [singleArticle, setSingleArticle] = useState([]);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    fetch(`https://sobe-news.herokuapp.com/api/articles/${article_id}`)
+      .then((res) => res.json())
+      .then(({ article }) => setSingleArticle(article));
+    setIsLoading(false);
+  }, [article_id]);
+
+  return (
+    <div className="single-article">
+      {isLoading && <div>Loading...</div>}
```

Comment:
You are getting an error "Cannot read properties of undefined" because you are only using `isLoading` to render the Loading div. You should use it to also check that it's finished loading before you render the whole component, so that line 30 (`singleArticle.created_at.slice....`) doesn't throw a typeError for trying to slice undefined.